lib/solar_system.rb

```diff
@@ -0,0 +1,34 @@
+class SolarSystem
+  attr_reader :star_name, :planets
+
+  def initialize(star_name)
+    @star_name = star_name
+    @planets = []
+  end
+
+  def add_planet(planet)
+    @planets << planet
+  end
+
+  def list_planets
+    numbered_planets_list = ""
+    planets.each_with_index do |planet, index|
+      numbered_planets_list += + "#{index+1}: #{planet.name}\n"
+    end
+    return "~~~~~~~~~~~~~~~~~~~~~~~~\nPlanets orbiting #{@star_name}\n#{numbered_planets_list}~~~~~~~~~~~~~~~~~~~~~~~~\n"
+  end
+
+  def find_planet_by_name(planet_name)
+    planet_object = planets.select {|planet| planet if planet.name.capitalize == planet_name.capitalize}
+    if planet_object != []
+      return planet_object
```

Comment:
You are returning an Array of planets not a single planet here.  You should use something like:  `planet_object.first`